components/UserItems.tsx

```
@@ -0,0 +1,62 @@
+import Item from "@models/item";
+import { connectToDB } from "@utils/database";
+import LogoutButton from "./LogoutButton";
+import User from "@models/user";
+import { IItem, IUser } from "@types";
+import ItemsWrapper from "./ItemsWrapper";
+import HeaderTitle from "./HeaderTitle";
+import Link from "next/link";
+
+export default async function UserItems({ session }: { session: any }) {
+  let items: IItem[] = [];
+  let user!: IUser;
+
+  try {
+    await connectToDB();
+    items = await Item.find({}).lean();
+    user = (await User.findById(session.id).lean()) as IUser;
```

for example:

    const userFromDB = await User.findById(session.id).lean();
    if (!userFromDB) {
      throw new Error("User not found");
    }

    user = userFromDB.toObject();
  } catch (error) {

---

better to `fail fast` and throw error if type/object is not in expected shape